oceanprotocol/RBAC-Server
# Issue 43 provider access control

Closes #43 

Changes proposed in this PR:

- Checking if users is allowed to use the provider url on all publish requests
- Requesting provider check from new API end point 
- Adding option to check from json env for when RBAC is running in barge

## To test

Run the RBAC locally on this branch `npm run build && npm run start` then send the following request to `http://localhost:3000` from postman:

```JSON
{ 
     "eventType": "publish", 
     "component": "market" ,
     "authService": "json",
     "did": "did:op:725ab48b618389Bf171533D07f9eBBB6ED0273Ee",
     "providerAddress": "0xf31FC0D18A24c6Ae4225A0d4eEB709AC0a18E993",
     "credentials": {
              "type": "address",
              "value": "0xB149C36986cf8e2B260d2Eb02AFf97058a1a5C95"
     }
}
```

You should get a `true` response
